File: practica1/Ejercicio3/src/ejercicio3/EmpleadoBuilder.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ejercicio3;

/**
 *
 * @author carme
 */
public abstract class EmpleadoBuilder {
    protected Empleado empleado;

    
    public EmpleadoBuilder()
    {
        empleado = new Empleado("", "", "", "");
    }

    public abstract EmpleadoBuilder build(String nombre, String dni, String cargo);
    
    public Empleado getEmpleado() {
        return empleado;
    }
}

File: practica2/ejercicio_grupal/lib/Model/empleadoBuilder.dart
import 'package:ejercicio_grupal/Model/elementoEmpresa.dart';

import 'empleado.dart';
import 'elementoEmpresa.dart';

abstract class EmpleadoBuilder {
    late Empleado empleado;

    EmpleadoBuilder(ElementoEmpresa? DepSuperior)
    {
        empleado = new Empleado.vacio();
    }

    EmpleadoBuilder build(String nombre, String dni, String cargo, ElementoEmpresa? DepSuperior);

    Empleado getEmpleado(){
        return Empleado(empleado.toString(), empleado.getDni(), empleado.getCargo(), empleado.getTipoContrato(), empleado.getSuperior());
    }

}

